### ISSUE TITLE
Worse performance compared with LISA

### ISSUE THREAD

**Issue by shupinghu** [MaverickRen/PixelLM#13]
Thank you for your released code and weights! I'm trying to deploy the released model on V100 recently, but the seg performance is worse than LISA. For example, I use the prompt "find the person in the picture" to seg the same picture, the output of PixelLM is:
![q78ZpawfjW](https://github.com/MaverickRen/PixelLM/assets/52944151/f7a4f594-b032-4a47-95ce-1c52c16a6672)
The output of LISA is:
![Fd4jOoDOKd](https://github.com/MaverickRen/PixelLM/assets/52944151/fead7dc8-6fd0-4d6e-89ca-34d8acadd68e)
Note:
1. For PixelLM, we use "PixelLM-7B" float32 model. If we use bf16 model, the error ""upsample_nearest2d_out_frame" not implemented for 'BFloat16'" occured; If we use fp16 model, the error "'LlamaAttention' object has no attribute 'rope_theta'" occured
2. For LISA, we use 7B bf16 model.

versions: 
Python: 3.8.0
Pytorch: 2.0.0+cu17
transformers: 4.31.0
tokenizers: 0.13.3
deepspeed： 0.14.0
openai： 0.27.8

**Comment by shupinghu** [comment]
Annother comparison:
in pixelLM:
![55CGCiEV4J](https://github.com/MaverickRen/PixelLM/assets/52944151/b26f341e-2f38-4823-86b8-205a8ba42938)
in LISA:
![2aZthRCPRC](https://github.com/MaverickRen/PixelLM/assets/52944151/71bab72c-879c-4699-af20-5d6ee5239b6e)

**Comment by MaverickRen** [comment]
Although PixelLM uses a lightweight decoder, it can do equivalent or even better work than using SAM on data sets such as refcoco, such as LISA. However, SAM is still more stable in the fine segmentation of complex scenes. This is related to the high-resolution input of SAM, and we are also working on solving this problem

**Comment by shupinghu** [comment]
> Although PixelLM uses a lightweight decoder, it can do equivalent or even better work than using SAM on data sets such as refcoco, such as LISA. However, SAM is still more stable in the fine segmentation of complex scenes. This is related to the high-resolution input of SAM, and we are also working on solving this problem

So it means that pixelLM could have a better performance in refcoco as shown in paper, but pixelLM could not have a better performance in coco? 

Since the reason is that SAM used a high-resolution input, what if we use a high-resolution input as well? According to "chat.py", the "image_size" is set to 1024, "resize_vision_tower_size" is set to 448, when I set "image_size=1920", "resize_vision_tower_size=1920", the infer results have no differences.

**Comment by MaverickRen** [comment]
The openai-CLIP encoder we use is trained on 336 pixels. Even if you increase it to 1920, it will not be further helpful.

**Comment by MaverickRen** [comment]
According to experience, the scenes in the COCO dataset will be more cluttered, and training at a higher resolution is helpful for this. But this will also bring a higher training load

**Comment by shupinghu** [comment]
Despite the input size of openai-CLIP, do you think that there are other reasones that maybe cause the poor performance on segment edge? For example, less training dataset?

**Comment by MaverickRen** [comment]
Yes, our training volume of split data (~200k) is nowhere near that of SAM. Maybe you can learn about ViT-adapter, which can optimize this problem

**Comment by shupinghu** [comment]
As I know, the training dataset that PixelLM used is the same as LISA, but because LISA uses SAM as the vision backbone (freeze the parameters) that use much more trainging dataset, this is the another reason that the performance of LISA is better, am I right?

When it comes to ViT-adapter, this is kind of finetuning method. Do you mean that I can use this method to finetune the released PixelLM-7B model in a bigger input size on my own dataset (or on the whole trainging dataset as in your paper)? If yes, maybe I also need change "--vision-tower='openai/clip-vit-large-patch14-336' " and "--resize_vision_tower_size=448" to a bigger one during training?
